.dotnet/src/Custom/Chat/ChatResponseFormat.cs

```diff
@@ -18,28 +20,49 @@ namespace OpenAI.Chat;
 /// the model.
 /// </para>
 /// </remarks>
-[CodeGenModel("CreateChatCompletionRequestResponseFormat")]
-public partial class ChatResponseFormat
+[CodeGenModel("ChatResponseFormat")]
+public abstract partial class ChatResponseFormat
 {
-    // CUSTOM: Made internal.
-
-    /// <summary> Must be one of `text` or `json_object`. </summary>
-    [CodeGenMember("Type")]
-    internal InternalCreateChatCompletionRequestResponseFormatType? Type { get; set; }
+    /// <summary> text. </summary>
+    public static ChatResponseFormat Text { get; } = new InternalChatResponseFormatText();
+    /// <summary> json_object. </summary>
+    public static ChatResponseFormat JsonObject { get; } = new InternalChatResponseFormatJsonObject();
 
-    // CUSTOM: Made internal.
-    /// <summary> Initializes a new instance of <see cref="ChatResponseFormat"/>. </summary>
-    internal ChatResponseFormat()
+    public static ChatResponseFormat FromJsonSchema(
+        string name,
+        string description = null,
+        BinaryData jsonSchema = null,
```

Comment:
The API reference says this is optional, but we confirmed that the service actually requires it, correct? Is this true for both, when strict = true and when strict = false?

Reply:
Fails with true, false, and omitted.

```
    System.ClientModel.ClientResultException : HTTP 400 (invalid_request_error: missing_required_parameter)
Parameter: response_format.json_schema.schema

Missing required parameter: 'response_format.json_schema.schema'.
```
In light of that, I'll update to make it a required parameter. If it's behaviorally loosened in the future to match the spec, we can just update the optionality to match without it being a breaking change. Not true the other way around.